tests/adapters/switches/juniper_test.py

```diff
@@ -3622,6 +3622,165 @@ def test_unset_interface_native_vlan_on_default_interface_raises(self):
         with self.assertRaises(NativeVlanNotSet):
             self.switch.unset_interface_native_vlan("ge-0/0/6")
 
+    def test_unset_interface_auto_negotiation_state_works_when_enabled(self):
+        self.netconf_mock.should_receive("get_config").with_args(source="candidate", filter=is_xml("""
+                <filter>
+                  <configuration>
+                    <interfaces>
+                      <interface>
+                        <name>ge-0/0/6</name>
+                      </interface>
+                    </interfaces>
+                  </configuration>
+                </filter>
+            """)).and_return(a_configuration("""
+                <interfaces>
+                  <interface>
+                    <name>ge-0/0/6</name>
+                    <ether-options>
+                      <auto-negotiation/>
+                    </ether-options>
+                  </interface>
+                </interfaces>
+            """))
+
+        self.netconf_mock.should_receive("edit_config").once().with_args(target="candidate", config=is_xml("""
+                <config>
+                  <configuration>
+                    <interfaces>
+                      <interface>
+                        <name>ge-0/0/6</name>
+                        <ether-options>
+                          <auto-negotiation operation=\"delete\"/>
+                        </ether-options>
+                      </interface>
+                    </interfaces>
+                  </configuration>
+                </config>
+            """)).and_return(an_ok_response())
+
+        self.switch.unset_interface_auto_negotiation_state("ge-0/0/6")
+
+    def test_unset_interface_auto_negotiation_state_does_nothing_on_default_interface(self):
+        self.netconf_mock.should_receive("get_config").with_args(source="candidate", filter=is_xml("""
+                <filter>
+                  <configuration>
+                    <interfaces>
+                      <interface>
+                        <name>ge-0/0/6</name>
+                      </interface>
+                    </interfaces>
+                  </configuration>
+                </filter>
+            """)).and_return(a_configuration("""
+                <interfaces>
```

Comment:
the correct result here is an empty string, no <interface> if empty and not <interfaces> if empty, <configuration> will remain